### DISCUSSION TITLE
Custom Workflow Question - How to Use the Selection Node in ComfyUI

### DISCUSSION THREAD

**Discussion by patrickchamp** [Acly/krita-ai-diffusion#1373] [Q&A]
I'm trying to understand how to use the Krita Selection node in ComfyUI workflows. It outputs a mask, but my workflow needed an image as an input and I'm new to masks so I wasn’t clear on how to make use of the mask to get the desired image selection for use as an input Image. I was working with an 'Expression Editor' node that only accepts an image as the input, but I'm looking to understand how to do this in general.

**Attempts So Far:**

1. **Using Compositing Nodes**:  
   I tried the "Cut Mask to Image" node to merge the canvas image with the mask, but encountered errors like:
   ```
   RuntimeError: Given groups=1, weight of size [16, 3, 3, 3], expected input[1, 4, 640, 640] to have 3 channels, but got 4 channels instead.
   ```

2. **Krita Layer Node**:  
   I abandoned the Selection Node to see if I could do it with the Layer node. I created a layer with only the selected face, this worked, but instead of receiving a transparent layer with a face in Krita, like when you receive  inpainting previews of your selected area, I received a non transparent layer the size of the canvas which contained the face surrounded by white.

**Question**  
What’s the best approach for using the Selection node to pass an area of an image through a workflow, and return just the selection to ComfyUI, like you would when inpainting?

**Comment by hoverkraft** [comment]
Maybe take a look at the "LayerStyle" nodes "Crop by Mask" and "Restore Crop Box", that's what I use. The error you are getting is probably because you are trying to combine images or latents that are nto the same size. If you take care to keep image aspect in your workflow you can just resize to the right size before pasting back the image with restore crop box. I use the resize node from KJ Nodes for that.

**Comment by nabileisa** [comment]
Hi, from my experimentation you can do it  with either `canvas` or a `layer` using the `apply mask to image` node witch is available in the `comfyui-tooling-nodes` package , this will give you a masked image you can use.
![workflow](https://github.com/user-attachments/assets/f3986f08-2f8f-4229-adf1-7593ed000e4b)
